fluent-plugin-enhance-k8s-metadata/lib/fluent/plugin/filter_enhance_k8s_metadata.rb

```diff
@@ -113,7 +115,33 @@ def normalize_param
         log.debug "bearer_token_file: '#{@bearer_token_file}', exist: #{File.exist?(@bearer_token_file)}"
 
         @cache_ttl = :none if @cache_ttl <= 0
-        log.info "cache_ttl: #{cache_ttl}, cache_size: #{@cache_size}"
+        log.info "cache_ttl: #{@cache_ttl}, cache_size: #{@cache_size}, cache_refresh: #{@cache_refresh}"
+      end
+
+      def start_cache_timer
+        timer_execute(:"cache_refresher", @cache_refresh) {
+          entries = @cache.to_a
+          log.info "Refreshing metadata for #{entries.count} entries"
+
+          entries.each { |entry|
+            begin
+              log.debug "Refreshing metadata for key #{entry[0]}"
+              split = entry[0].split("::")
+              namespace_name = split[0]
+              pod_name = split[1]
+              metadata = fetch_pod_metadata(namespace_name, pod_name)
+
+              if !metadata.nil? && !metadata.empty?
+                @cache[entry[0]] = metadata
+              else
+                log.info "Deleting entry for key #{entry[0]}"
+                @cache.delete(entry[0])
```

Comment:
I don't think we should delete the entry here - the pod itself might be deleted, but we might still have logs or metrics to collect for that pod due to backpressuring or something. I think we can just let it expire on its own with the cache TTL

Reply:
that's a great point. I guess I can still check for an empty metadata after looking up the deleted pod, but just not update the entry in the cache so that it can expire on its own